prompting/datasets/wiki.py

```diff
@@ -217,9 +217,12 @@ def search(self, name, results=3) -> Context:
 
     def random(self, pages=10) -> dict:
         titles = _get_random_titles(pages=pages)
-        title = random.choice(titles)
-        return self.get(title)
-
+        for i in range(self.max_tries):
+            title = titles[i]
+            context = self.get(title)
+            if context:
+                return context
+        return None
```

Comment:
so this can be written a little more cleanly as

```
for title in titles[:self.max_tries]:
    if context := self.get(title):
        return context
```
most of this is purely visual and doesn't actually matter, but when self.max_tries is >= than len(titles) your code would throw an exception which isn't getting caught, so there is also an actual functional difference:)